clash-prelude/src/Clash/Explicit/Reset.hs

```diff
@@ -52,6 +55,53 @@ import           GHC.TypeLits (type (+))
 >>> import Clash.Explicit.Prelude
 -}
 
+-- | A deasserted reset
+noReset :: KnownDomain dom => Reset dom
+noReset = unsafeFromHighPolarity (pure False)
+
+-- | Output reset will be asserted when either one of the input resets is
+-- asserted
+orReset ::
+  forall dom .
+  ( KnownDomain dom
+  , DomainResetKind dom ~ 'Synchronous ) =>
```

Comment:
I don't think it's a good idea to solve this with a constraint because it'll need to be copied to all domain-polymorphic encompassing user code. Rather, let's do what we do in `clash-cores:Clash.Cores.Xilinx.DcFifo` and `resetGlitchFilter`:
```haskell
orReset = case resetKind @dom of
            SSynchronous -> unsafeOrReset
            SAsynchronous -> clashCompileError "Computer says No"
```

Reply:
Fair

Reply:
We had a long discussion about this in the office, and I think the conclusion was that we do want this at the type level.
But we want to make it more convenient to write `DomainResetKind dom ~ 'Synchronous`, with something like:
```haskell
type KnownSyncDomain (dom :: Domain) = (KnownDomain dom, DomainResetKind dom ~ 'Synchronous)

type DomIsSync (dom :: domain) = DomainResetKind dom ~ 'Synchronous
```